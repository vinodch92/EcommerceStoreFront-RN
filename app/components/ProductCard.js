import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import FastImage from 'react-native-fast-image';
import { images } from '../assets';
import config from '../config';

/**
 * This is ProductCard functional component for showing a product in a card
 * @param {*} productData : object of a product
 */
const ProductCard = ({ productData }) => {
  return (
    <View style={styles.cardContainer}>
      <View style={styles.cardContent}>
        <FastImage
          resizeMode="cover"
          style={styles.productImage}
          source={{ uri: productData?.image }}
          fallback
          defaultSource={images.placeholder}
        />
        <Text style={styles.productName}>{`${productData?.name ?? ''}`}</Text>
        <Text style={styles.productRating}>{`Rating: ${productData?.rating ?? 0}`}</Text>
      </View>
    </View>
  );
};

export default ProductCard;

const styles = StyleSheet.create({
  cardContainer: {
    borderRadius: 6,
    elevation: 3,
    backgroundColor: config.colors.white,
    shadowOffset: { width: 1, height: 1 },
    shadowColor: config.colors.cardShadow,
    shadowOpacity: 0.3,
    shadowRadius: 2,
    marginHorizontal: 20,
    marginVertical: 20,
  },
  cardContent: { marginHorizontal: 15, marginVertical: 10 },
  productImage: { width: '100%', height: 150, alignSelf: 'center' },
  productName: { marginTop: 15, alignSelf: 'center', fontSize: 17, fontWeight: '600' },
  productRating: {
    marginTop: 15,
    fontSize: 14,
    color: config.colors.ratingGreen,
    fontWeight: '400',
  },
});
