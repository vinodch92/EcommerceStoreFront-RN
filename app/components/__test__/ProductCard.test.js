import React from 'react';
import { shallow } from 'enzyme';
import { Text, View } from 'react-native';
import FastImage from 'react-native-fast-image';
import ProductCard from '../ProductCard';

describe('ProductCard', () => {
  describe('Rendering', () => {
    const wrapper = shallow(<ProductCard />);
    it('renders component correctly', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(2);
      expect(wrapper.find(Text).length).toEqual(2);
      expect(wrapper.find(FastImage).length).toEqual(1);

      wrapper.setProps({ productData: { name: 'Brown eggs' } });
      expect(wrapper.find(View).length).toEqual(2);
      expect(wrapper.find(Text).length).toEqual(2);
      expect(wrapper.find(FastImage).length).toEqual(1);

      wrapper.setProps({
        productData: {
          name: 'Brown eggs',
          image:
            'https://raw.githubusercontent.com/vinodch92/ECommerceProductImages/master/images/storeFront/0.jpg',
        },
      });
      expect(wrapper.find(View).length).toEqual(2);
      expect(wrapper.find(Text).length).toEqual(2);
      expect(wrapper.find(FastImage).length).toEqual(1);

      wrapper.setProps({
        productData: {
          name: 'Brown eggs',
          image:
            'https://raw.githubusercontent.com/vinodch92/ECommerceProductImages/master/images/storeFront/0.jpg',
          rating: 4,
        },
      });
      expect(wrapper.find(View).length).toEqual(2);
      expect(wrapper.find(Text).length).toEqual(2);
      expect(wrapper.find(FastImage).length).toEqual(1);
    });
  });
});
