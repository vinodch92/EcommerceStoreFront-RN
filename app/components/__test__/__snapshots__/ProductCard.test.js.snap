// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Rendering renders component correctly 1`] = `ShallowWrapper {}`;
